include/class.ticket.php

```diff
@@ -2624,6 +2624,17 @@ function changeOwner($user) {
     function postMessage($vars, $origin='', $alerts=true) {
         global $cfg;
 
+        // Validate Form Token to prevent form resubmissions
+        if (!$vars['form_token'] || !$vars['lockCode'])
+            $errors['err'] = __('Invalid Form Token');
+
```

Comment:
Validation shouldn't happen inside any model... it's a `form` validation, not part of data validation.